components/lark-type-check/src/hir_typeck.rs

```diff
@@ -394,7 +394,7 @@ where
                 let signature = self.substitute(expression, &generics, signature_decl);
 
                 // Relate the owner type to the input
-                self.equate(expression, expression, owner_access_ty, signature.inputs[0]);
+                self.equate(expression, expression, owner_access_ty, owner_access_ty);
```

Comment:
This was right before, I think. Equating a type to itself is a no-op. The idea here is that we want to equate the actual receiver type (`owner_access_ty`) with the type from the method signature (`signature.inputs[0]`).